src/components/progressCircular/progress-circular.js

```diff
@@ -87,43 +89,76 @@ function MdProgressCircularDirective($mdConstant, $mdTheming, $mdUtil) {
   function postLink(scope, element, attr) {
     $mdTheming(element);
 
-    var circle = element[0];
+    var circle = element;
     var spinnerWrapper =  angular.element(element.children()[0]);
-
     var lastMode, toVendorCSS = $mdUtil.dom.animator.toCss;
 
-    // Update size/scaling of the progress indicator
-    // Watch the "value" and "md-mode" attributes
-
-    circle.style[$mdConstant.CSS.TRANSFORM] = 'scale(' + getDiameterRatio() + ')';
-
-    attr.$observe('value', function(value) {
-      var percentValue = clamp(value);
-      element.attr('aria-valuenow', percentValue);
+    updateScale();
+    validateMode();
+    watchAttributes();
 
-      if (attr.mdMode == "determinate") {
-        animateIndicator(percentValue);
-      }
-    });
+    /**
+     * Watch the value and md-mode attributes
+     */
+    function watchAttributes() {
+     attr.$observe('value', function(value) {
+           var percentValue = clamp(value);
+           element.attr('aria-valuenow', percentValue);
+
+           if (mode() == MODE_DETERMINATE) {
+             animateIndicator(percentValue);
+           }
+         });
+     attr.$observe('mdMode',function(mode){
+       switch( mode ) {
+         case MODE_DETERMINATE:
+         case MODE_INDETERMINATE:
+           spinnerWrapper.removeClass('ng-hide');
+
+           // Inject class selector instead of attribute selector
+           // (@see layout.js changes for IE performance issues)
+
+           if ( lastMode ) spinnerWrapper.removeClass( lastMode );
+                lastMode = "md-mode-" + mode;
```

Comment:
If you're not paying attention, the indentation makes this look like it is part of the above `if` conditional.

Also, doesn't it make the next line `if (lastMode)` irrelevant since we're specifically setting it here? I mean, even if `mode` is undefined, `last-mode` would still be `"md-mode-"` right?